File: src/main/java/com/ject/studytrip/trip/application/dto/TripReportInfo.java
package com.ject.studytrip.trip.application.dto;

import com.ject.studytrip.global.util.DateUtil;
import com.ject.studytrip.trip.domain.model.TripReport;

public record TripReportInfo(
        Long tripReportId,
        String title,
        String content,
        String startDate,
        String endDate,
        long completedMissionCount,
        long totalFocusHours,
        long studyDays,
        String imageTitle,
        String imageUrl,
        String createdAt,
        String updatedAt,
        String deletedAt) {
    public static TripReportInfo from(TripReport tripReport) {
        return new TripReportInfo(
                tripReport.getId(),
                tripReport.getTitle(),
                tripReport.getContent(),
                tripReport.getStartDate(),
                tripReport.getEndDate(),
                tripReport.getCompletedMissionCount(),
                tripReport.getTotalFocusHours(),
                tripReport.getStudyDays(),
                tripReport.getImageTitle(),
                tripReport.getImageUrl(),
                DateUtil.formatDateTime(tripReport.getCreatedAt()),
                DateUtil.formatDateTime(tripReport.getUpdatedAt()),
                DateUtil.formatDateTime(tripReport.getDeletedAt()));
    }
}

File: src/main/kotlin/com/ject/studytrip/trip/application/dto/TripReportInfo.kt
package com.ject.studytrip.trip.application.dto

import com.ject.studytrip.global.util.DateUtil
import com.ject.studytrip.global.util.EntityExtensions.requireId
import com.ject.studytrip.trip.domain.model.TripReport

data class TripReportInfo(
    val tripReportId: Long,
    val title: String,
    val content: String,
    val startDate: String,
    val endDate: String?,
    val studyLogCount: Long,
    val totalFocusHours: Long,
    val studyDays: Long,
    val imageTitle: String?,
    val imageUrl: String?,
    val createdAt: String,
    val updatedAt: String,
    val deletedAt: String?,
) {
    companion object {
        fun from(tripReport: TripReport): TripReportInfo =
            TripReportInfo(
                tripReport.id.requireId(),
                requireNotNull(tripReport.title),
                tripReport.content,
                tripReport.startDate,
                tripReport.endDate,
                tripReport.studyLogCount,
                tripReport.totalFocusHours,
                tripReport.studyDays,
                tripReport.imageTitle,
                tripReport.imageUrl,
                DateUtil.formatDateTime(requireNotNull(tripReport.createdAt)),
                DateUtil.formatDateTime(requireNotNull(tripReport.updatedAt)),
                tripReport.deletedAt?.let { DateUtil.formatDateTime(it) },
            )
    }
}
